grisafi/app.js

```diff
@@ -0,0 +1,269 @@
+            return socios[i];
+        }
+    }
+    return 0;
+}
+
+function adeuda(idSocio){
+    var fecha_prest=new Array();
+    for (let i = 0; i < prestamos.length; i++) {
+        if(idSocio==prestamos[i].idSocio){
+            fecha_prest.push(new Date(prestamos[i].fechavencimiento));
+        }
+    }
+    for (let i = 0; i < fecha_prest.length; i++) {
+        if(Date.now()>fecha_prest[i]){
+            return true;
+        }
+    }
+    return false;
+}
+
+function checkPrestado(idLibro){
+    for (let i = 0; i < prestamos.length; i++) {
+        if(prestamos[i].idLibro==idLibro){
+            return 1;
+        }   
+    }
+    return 0;
+}
+
+function deleteBook(id){
+    if(checkPrestado()==0){
+        for (let i = 0; i < libros.length; i++) {
+            if(id==libros[i].id){
+                libros.splice(i,1);
+                return 1;
+            }
+        }
+        return 0;
+    }
+    else{
+        return -1;
+    }
+}
+
+function getPrestamos(idSocio){
+    var prest=new Array();
+    for (let i = 0; i < prestamos.length; i++) {
+        if(idSocio==prestamos[i].idSocio){
+            prest.push({"idLibro":prestamos[i].idLibro ,"Fecha Vencimiento":new Date(prestamos[i].fechavencimiento)});
+        }
+    }
+    return prest;
+}
+
+function generatePrestamosID(){
+    return Math.random().toString(36).substr(2, 9);
+}
+
+libros.push(new Libro("Harry Potter",100,10));
+libros.push(new Libro("El Señor de los Anillos",10,20));
+
+socios.push(new Socio("A",1));
+socios.push(new Socio("B",2));
+
+prestamos.push(new Prestamo(1,1,10,20));
+prestamos.push(new Prestamo(2,2,10,20));
+
+app.get('/socios',function(req,res){
+    res.status(200).json(socios);
+})
+
+app.get('/socios/:idSocio',function(req,res){
+    var socio=findMember(req.params.idSocio);
+    if(socio!=0){
+        res.status(200).json(findMember(req.params.idSocio));
+    }
+    else{
+        res.status(404).send("Socio no encontrado");
+    }
+})
+
+app.get('/socios/:idSocio/prestamos',function(req,res){
+    if(findMember(req.params.idSocio)){
+        var prest=getPrestamos(req.params.idSocio);
+        if(prest.length==0){
+            res.status(200).send("No hay Prestamos Realizados por el Socio");
+        }
+        else{
+            res.status(200).json(prest);
+        }
+    }
+    else{
+        res.status(400).send("Socio No Encontrado");
+    }
+})
+
+app.get('/prestamos',function(req,res){
+    if(prestamos!=null)
+        res.status(200).json(prestamos);
+    else
+        res.status(204).send("No hay Prestamos Registrados");
+})
+
+app.get('/libros/:idLibro',function(req,res){
+    var book=findBook(req.params.idLibro);
+    if(book!=0){
+        res.status(200).json({"idLibro":parseInt(req.params.idLibro),"disponibles":book.disponibles()});
+    }
+    else{
+        res.status(400).send("Libro no encontrado");
+    }
+})
+
+app.post('/socios',function(req,res){
+    socios.push(new Socio(req.body.nombre,req.body.id));   
```

Comment:
You should check here that the values in the request body are not null or negative